tests/benchmarks.cpp

```diff
@@ -0,0 +1,148 @@
+#include "benchmark/benchmark.h"
+#include "gmock/gmock.h"
+
+#include "mocks.h"
+
+#include "ast/passes/codegen_llvm.h"
+#include "ast/passes/field_analyser.h"
+#include "ast/passes/resource_analyser.h"
+#include "ast/passes/semantic_analyser.h"
+
+#include "bpffeature.h"
+#include "bpftrace.h"
+#include "clang_parser.h"
+#include "driver.h"
+
+namespace bpftrace {
+namespace test {
+
+// We break the benchmarks into several phases, in order to track issues
+// separately. Note that currently each meaningful pass has its own phase, but
+// these could be logically merged as "optimizations" or some other passes with
+// some minor code restructuring.
+enum class Phase {
+  Parse,
+  FieldAnalyser,
+  ClangParser,
+  SemanticAnalyser,
+  ResourceAnalyser,
+  CodegenGenerateIR,
+  CodegenOptimize,
+  CodegenEmit,
+};
+
+constexpr const char *phaseName(Phase p)
+{
+  switch (p) {
+    case Phase::Parse:
+      return "parse";
+    case Phase::FieldAnalyser:
+      return "field_analyser";
+    case Phase::ClangParser:
+      return "clang_parser";
+    case Phase::SemanticAnalyser:
+      return "semantic_analyser";
+    case Phase::ResourceAnalyser:
+      return "resource_analyser";
+    case Phase::CodegenGenerateIR:
+      return "codegen_generate_ir";
+    case Phase::CodegenOptimize:
+      return "codegen_optimize";
+    case Phase::CodegenEmit:
+      return "codegen_emit";
+    default:
+      return "unknown";
```

Comment:
nit: should be ok to drop default. We have the compiler flags on to warn if switch is not exhaustive. Adding `default` makes it easier for bugs to slip in

Reply:
Done.